File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/model/mapper/ParkingZoneResponseMapper.java
package com.lattis.ellipse.data.network.model.mapper;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.network.base.AbstractDataMapper;
import com.lattis.ellipse.data.network.model.response.parking.GetParkingZoneDataResponse;
import com.lattis.ellipse.data.network.model.response.parking.ParkingZoneGeometryResponse;
import com.lattis.ellipse.domain.model.ParkingZone;
import com.lattis.ellipse.domain.model.ParkingZoneGeometry;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;


public class ParkingZoneResponseMapper extends AbstractDataMapper<GetParkingZoneDataResponse, ParkingZone> {

    @Inject
    public ParkingZoneResponseMapper() {
    }

    @NonNull
    @Override
    public ParkingZone mapIn(@NonNull GetParkingZoneDataResponse getParkingZoneDataResponse) {

        ParkingZone parkingZone = new ParkingZone();
        parkingZone.setCustomerID(getParkingZoneDataResponse.getCustomerID());
        parkingZone.setFleetID(getParkingZoneDataResponse.getFleetID());
        parkingZone.setParking_area_id(getParkingZoneDataResponse.getParking_area_id());
        parkingZone.setType(getParkingZoneDataResponse.getType());
        parkingZone.setZone_Name(getParkingZoneDataResponse.getZone_Name());
        parkingZone.setZone(getParkingZoneDataResponse.getZone());
        List<ParkingZoneGeometryResponse> responses = getParkingZoneDataResponse.getParkingZoneGeometry();
        List<ParkingZoneGeometry> parkingZoneGeometryList = new ArrayList<>();
        if (responses != null) {
            for (ParkingZoneGeometryResponse parkingZoneGeometryResponse : responses) {
                if (parkingZoneGeometryResponse != null) {
                    ParkingZoneGeometry parkingZoneGeometry = new ParkingZoneGeometry();
                    parkingZoneGeometry.setLatitude(parkingZoneGeometryResponse.getLatitude());
                    parkingZoneGeometry.setLongitude(parkingZoneGeometryResponse.getLongitude());
                    parkingZoneGeometry.setRadius(parkingZoneGeometryResponse.getRadius());
                    parkingZoneGeometryList.add(parkingZoneGeometry);
                }
            }
        }

        parkingZone.setParkingZoneGeometry(parkingZoneGeometryList);
        return parkingZone;
    }

    @NonNull
    @Override
    public GetParkingZoneDataResponse mapOut(@NonNull ParkingZone parkingZone) {
        return null;
    }

}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/mapper/ParkingZoneResponseMapper.kt
package com.lattis.data.mapper

import com.lattis.data.entity.response.parking.GetParkingZoneDataResponse
import com.lattis.data.entity.response.parking.ParkingZoneGeometryResponse
import com.lattis.domain.models.ParkingZone
import com.lattis.domain.models.ParkingZoneGeometry
import java.util.*
import javax.inject.Inject

class ParkingZoneResponseMapper @Inject constructor() :
    AbstractDataMapper<GetParkingZoneDataResponse, ParkingZone>() {

    override fun mapIn(getParkingZoneDataResponse: GetParkingZoneDataResponse?): ParkingZone {
        val parkingZone = ParkingZone()
        if(getParkingZoneDataResponse!=null) {
            parkingZone?.customerID = (getParkingZoneDataResponse?.customerID)
            parkingZone?.fleetID = (getParkingZoneDataResponse?.fleetID)
            parkingZone?.parking_area_id = (getParkingZoneDataResponse?.parking_area_id)
            parkingZone?.type = (getParkingZoneDataResponse?.type)
            parkingZone?.zone_Name = (getParkingZoneDataResponse?.zone_Name)
            parkingZone?.zone = (getParkingZoneDataResponse?.zone)
            val responses: List<ParkingZoneGeometryResponse>? =
                getParkingZoneDataResponse?.parkingZoneGeometry
            val parkingZoneGeometryList: MutableList<ParkingZoneGeometry> =
                ArrayList<ParkingZoneGeometry>()
            if (responses != null) {
                for (parkingZoneGeometryResponse in responses) {
                    if (parkingZoneGeometryResponse != null) {
                        val parkingZoneGeometry = ParkingZoneGeometry()
                        parkingZoneGeometry.latitude = (parkingZoneGeometryResponse.latitude)
                        parkingZoneGeometry.longitude = (parkingZoneGeometryResponse.longitude)
                        parkingZoneGeometry.radius = (parkingZoneGeometryResponse.radius)
                        parkingZoneGeometryList.add(parkingZoneGeometry)
                    }
                }
            }
            parkingZone.parkingZoneGeometry = (parkingZoneGeometryList)
        }
        return parkingZone
    }

    override fun mapOut(parkingZone: ParkingZone?): GetParkingZoneDataResponse? {
        return null
    }
}